src/dropdown_select.rs

```diff
@@ -0,0 +1,333 @@
+// Copyright 2019 The Druid Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! A simple list selection widget, for selecting a single value out of a list.
+
+use crate::{Dropdown, ListSelect, Wedge, DROP};
+use druid::kurbo::BezPath;
+use druid::widget::{
+    Click, Container, Controller, ControllerHost, DefaultScopePolicy, Label, LabelText, Scope,
+};
+use druid::{
+    theme, Affine, BoxConstraints, Data, Env, Event, EventCtx, Insets, LayoutCtx, Lens, LifeCycle,
+    LifeCycleCtx, LinearGradient, PaintCtx, Point, RenderContext, Selector, Size, UnitPoint,
+    UpdateCtx, Widget, WidgetExt, WidgetPod,
+};
+use std::marker::PhantomData;
+
+// NOTE: This is copied from Button. Should those be generic, or maybe set in the environment?
+const LABEL_INSETS: Insets = Insets::uniform_xy(8., 2.);
+const COLLAPSE: Selector<()> = Selector::new("druid-widget-nursery.dropdown.collapse");
+
+/// A list selection widget, showed as a button, for which the different possible values appear as a dropdown.
+pub struct DropdownSelect<T> {
+    _t: PhantomData<T>,
+}
+
+impl<T: Data + PartialEq> DropdownSelect<T> {
+    /// Given a vector of `(label_text, enum_variant)` tuples, create a dropdown select widget
+    /// This is exactly the same interface as `Radio` so that both can be used interchangably,
+    /// with dropdown taking less space in the UI.
+    pub fn new(
```

Comment:
It feels like 'new' is the wrong name here because we aren't returning an instance of DropDownSelect (which seems to be 'just' a controller?) 
I think we are missing some stuff to be able to conveniently declare this kind of composite widget. (Tabs and druid-table go to great lengths to wrap their internals up in a widget rather than return a bare Scope....) . Maybe some macro but I'm not sure.

Reply:
I could easily rename this `build` or `build_widget`. Would it be better?
I'll check how Tabs and druid-table handle this.

Reply:
I'm not sure the way that they do it is necessarily something to spread around too much at the moment (which boils down to another wrapper just to hide their contents) - maybe this is a pattern we want to make easier in the end. In my mind it should be nearly as easy as declaring a component in a reactive web framework like Vue - modulo Rust vs JS/TS issues. Definitely needs a bit of thought....

I think build_widget might be a decent idea. I don't think I currently have a decent idea about scoping, but it does seem slightly weird that it is within a struct that happens to be used as a Controller deeper in the graph.

Reply:
I think the best solution for now is to split `ListSelect` into the empty shell doing the `build_widget` on one side, and a `ListSelectController` handling the up/down arrows for changing selection on the other side. That's both an easy and proper way to handle this. I'll also change the comment to `ListSelect` so it does not claim to *be* a widget, but rather to *build* one. Same for `DropdownSelect` which has the same architecture.
I'll include that in my next PR.

Reply:
Makes sense to me, the controller is really just an implementation detail. I think the "right" patterns will take a while to emerge...